keylime/tenant.py

```diff
@@ -743,36 +748,39 @@ def do_cvdelete(self, verifier_check=True):
         do_cvdelete = RequestsClient(self.verifier_base_url, True, tls_context=self.tls_context)
         response = do_cvdelete.delete(f"/v{self.api_version}/agents/{self.agent_uuid}", timeout=self.request_timeout)
 
-        response = response.json()
+        response_json = Tenant._jsonify_response(response, True)
 
-        if response["code"] == 503:
+        if response_json["code"] == 503:
             logger.error(
                 "Cannot connect to Verifier at %s with Port %s. Connection refused.",
                 self.verifier_ip,
                 self.verifier_port,
             )
             return response
-        if response["code"] == 504:
+        if response_json["code"] == 504:
             logger.error("Verifier at %s with Port %s timed out.", self.verifier_ip, self.verifier_port)
             return response
-        if response["code"] == 202:
+        if response_json["code"] == 202:
             deleted = False
             for _ in range(12):
                 get_cvdelete = RequestsClient(self.verifier_base_url, True, tls_context=self.tls_context)
                 response = get_cvdelete.get(
                     f"/v{self.api_version}/agents/{self.agent_uuid}", timeout=self.request_timeout
                 )
 
+                response_json = Tenant._jsonify_response(response, False, True)
```

Comment:
Here you convert it a 2nd time...

Reply:
It is a new response. This section of the code is a total mess, and I intend to fix that in the next PR :-)

Reply:
ah, yes, right.